src/main/java/com/faforever/api/security/AuditService.java

```diff
@@ -9,16 +9,27 @@
 @Service
 public class AuditService {
 
-  private UserSupplier userSupplier;
+  private final UserSupplier userSupplier;
 
   public AuditService(UserSupplier userSupplier) {
     this.userSupplier = userSupplier;
   }
 
   public void logMessage(String message) {
-    final FafAuthenticationToken fafAuthenticationToken = userSupplier.get();
-    String extendedMessage = MessageFormat.format("{0} [invoked by User ''{1}'' with id ''{2}'']",
-      message, fafAuthenticationToken.getUsername(), fafAuthenticationToken.getUserId());
+    final String extendedMessage = userSupplier.get()
+      .map(fafAuthenticationToken -> {
+        //move to switch pattern matching with java 21
+        if (fafAuthenticationToken instanceof FafUserAuthenticationToken fafUserAuthenticationToken) {
+          return MessageFormat.format("{0} [invoked by User ''{1}'' with id ''{2}'']",
+            message, fafUserAuthenticationToken.getUsername(), fafUserAuthenticationToken.getUserId());
+        } else if (fafAuthenticationToken instanceof FafServiceAuthenticationToken fafServiceAuthenticationToken) {
+          return MessageFormat.format("{0} [invoked by Service ''{1}'']",
+            message, fafServiceAuthenticationToken.getServiceName());
+        } else {
+          throw new RuntimeException();
```

Comment:
Message "Unknown authentication token type"

Reply:
ok, but this should not happen with sealed class, right?

Reply:
We aren't on java 21 yet XD

Reply:
We are on develop now :P